Improvements and new features

abougouffa opened the pull request:
Hello @bbatsov 

Thank you for this package.

This PR adds:

- Refactor some parts of the code using Emacs built-in functions (`seq-some` and `seq-every-p`)
- Adds `super-save-silent` to auto save files silently
- Adds `super-save-delete-trailing-whitespaces` which accepts `nil` (default), `t` to call `delete-trailing-whitespace` before saving. The variable also accepts a symbol `except-current-line`, which delete trailing white spaces for all lines except the current line. This is useful when we are in the middle of writing something, but we need to switch to something else for a while. In this case, we would expect to find the cursor where we left it.

The two last features are inspired by https://github.com/manateelazycat/auto-save

abougouffa committed:
Refactor and simplify the code

abougouffa committed:
Add `super-save-silent` to save files silently

abougouffa committed:
Option to delete trailing whitespaces (opt. except the current line)

bbatsov commented on super-save.el:
```diff
@@ -5,7 +5,7 @@
 ;; URL: https://github.com/bbatsov/super-save
 ;; Keywords: convenience
-;; Version: 0.4.0-snapshot
+;; Version: 0.4.1-snapshot
```
There's no need to bump a snapshot version.

bbatsov commented on super-save.el:
```diff
@@ -66,16 +69,28 @@ See `super-save-auto-save-when-idle'."
+  "Save silently, don't display any message."
+  :group 'super-save
+  :type 'boolean
+  :package-version '(super-save . "0.4.1"))
```
Those versions should stay at 0.4, as it has not been released yet.

bbatsov commented:
You should also mention the new configuration options in the README. And check in which version of Emacs exactly `seq.el` was introduced, as we might have to bump the Emacs 24.4 requirement.

abougouffa commented:
Hello @bbatsov

Thank you for the feedback, I've addressed the comments you raised in the last commit. [`seq.el` was introduced in Emacs 25.1](https://github.com/NicolasPetton/seq.el?tab=readme-ov-file#note-package-now-in-emacs--elpa), I've bumped the dependency in the package's documentation.

abougouffa committed:
Address merge comments

bbatsov commented:
Thanks!
